src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcEndpointDataSource.cs

```diff
@@ -222,18 +222,40 @@ private void UpdateEndpoints()
             bool suppressPathMatching)
         {
             var newPathSegments = routePattern.PathSegments.ToList();
+            int? requiredParameterEndpointRouteOrder = null;
 
             for (var i = 0; i < newPathSegments.Count; i++)
             {
                 // Check if the pattern can be shortened because the remaining parameters are optional
                 //
-                // e.g. Matching pattern {controller=Home}/{action=Index}/{id?} against HomeController.Index
-                // can resolve to the following endpoints:
-                // - /Home/Index/{id?}
-                // - /Home
+                // e.g. Matching pattern {controller=Home}/{action=Index} against HomeController.Index
+                // can resolve to the following endpoints: (ordered by precidence)
```

Comment:
`precidence` precedence

Reply:
Also `ordered by precedence` isn't what this PR is doing - precedence is a purely computed value

Reply:
`(sorted by RouteEndpoint.Order)`?